### PR TITLE
Add submission navigation to the show submission page

### PR DESCRIPTION
**Pull request by nzcodarnoc**
<details>
<summary>screen capture</summary>

image
</details>

### CODE REVIEW

File: app/controllers/submissions_controller.rb [hunk 1]
```diff
@@ -112,8 +112,16 @@ def mark_restored
...
   private
 
+  def submission_id_from_route
+    if params[:id] == "goto"
+      params[:goto_submission_id]
+    else
+      params[:id] || params[:submission_id]
+    end
+  end
+
```
Comment (nzcodarnoc) [inline app/controllers/submissions_controller.rb:122]: This doesn't seem right to me. I'm sure it should be a change in the router, but, I don't know enough about the router to do the change.

File: app/models/submission.rb [hunk 2]
```diff
@@ -126,20 +125,20 @@ def survey_id
     survey&.id
   end
 
-  def next
-    consultation.submissions.navigable.order(:created_at).find_by("id > ?", id)
+  def next(states)
```
Comment (mermop) [inline app/models/submission.rb:128]:
```suggestion
  def next(states: STATES)
```

File: app/views/submissions/show.html.erb [hunk 3]
```diff
@@ -1,5 +1,18 @@
 <div class="row">
-  <div class="col text-right">
+  <div class="col mb-3">
+    <%= render partial: "submission_navigation", locals: {
+      first_url: submission_path(@submission.first(%w[incoming ready started finished])),
+      prev_url: @submission.prev(%w[incoming ready started finished]) ? submission_path(@submission.prev(%w[incoming ready started finished])) : "#",
```
Comment (mermop) [inline app/views/submissions/show.html.erb:5]:
```suggestion
      prev_url: @submission.prev(states: %w[incoming ready started finished]) ? submission_path(@submission.prev(%w[incoming ready started finished])) : "#",
```

File: app/models/submission.rb [hunk 4]
```diff
@@ -29,7 +29,6 @@ class Submission < ApplicationRecord # rubocop:disable Metrics/ClassLength
   delegate :blank?, to: :text, prefix: true
```
Comment (mermop) [inline app/models/submission.rb:30]:
```suggestion
TAGGABLE_STATES = %w[ready started]
SHOWABLE_STATES = %w[incoming ready started finished]
EDITABLE_STATES = %w[incoming ready started]
```

File: app/views/submissions/tag.html.erb [hunk 5]
```diff
@@ -14,7 +14,15 @@
   </section>
 
   <section class="col-6 offset-3">
-    <%= render partial: "submission_navigation", locals: { submission: @submission } %>
+    <%= render partial: "submission_navigation", locals: {
+      first_url: consultation_submission_tag_path(@submission.consultation, @submission.first(%w[ready started])),
```
Comment (mermop) [inline app/views/submissions/tag.html.erb:18]:
```suggestion
      first_url: consultation_submission_tag_path(@submission.consultation, @submission.first(states: Submission::TAGGABLE_STATES)),
```

**Pull request closed without merging**